### DISCUSSION TITLE
peakrdl --help gives "peakrdl: command not found"

### DISCUSSION THREAD

**Discussion by martin-tanguay** [SystemRDL/systemrdl-compiler#236] [Q&A] [answered]
Hi,

I've installed peakrdl on Ubuntu 24.04 using these commands :

> sudo apt-get install python3-pip 
> python3 -m pip install peakrdl --break-system-packages 

When I run the command "peakrdl --help" on the terminal, it gives me : peakrdl: command not found

Looking at the installation folder, peakrdl is located in /home/username/.local/bin, and the packages are in /home/usrename/.local/lib/python3.12/site-packages

So going in the folder /home/username/.local/bin, I can run this command :

_~/.local/bin $ python3 peakrdl --help
usage: peakrdl [-h] [--version] [--plugins] [-f FILE] [--peakrdl-cfg CFG] <subcommand> ...

PeakRDL is a control & status register model automation toolchain.

For help about a specific subcommand, try:
    peakrdl <command> --help

For more documentation, visit https://peakrdl.readthedocs.io

options:
  -h, --help         show this help message and exit
  --version          show program's version number and exit
  --plugins          Report the PeakRDL plugins, their versions, then exit
  -f FILE            Specify a file containing more command line arguments
  --peakrdl-cfg CFG  Specify a PeakRDL configuration TOML file

subcommands:
    dump             print register model contents to stdout
    globals          list all globally accessible types that can be elaborated as top
    preprocess       Preprocess SystemRDL and write the result to a file
    uvm              Generate a UVM register model
    html             Generate HTML documentation
    c-header         Generate a C header definition of an address space
    regblock         Generate a SystemVerilog control/status register (CSR) block
    ip-xact          Export the register model to IP-XACT
    systemrdl        Export the register model to SystemRDL_

Any idea why I need to run it from the peakrdl folder with the python3 prefix ?

Thanks !

Martin

**Comment by amykyta3** [comment, marked as answer]
Add this to `~/.bashrc` or `~/.profile`:
```
# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
```

When you installed peakrdl, the pip installer probably warned you that this was missing. Likely something like:
```
  WARNING: The script peakrdl is installed in '/home/alex/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
```

**Reply by martin-tanguay** [reply]
That worked.  Thanks a lot for your quick feedback !
